Documentation/planning/arcade-powered-source-build/intermediate-nupkg.md

```diff
@@ -35,14 +35,90 @@ source-built intermediates.
 
 ## Too large?
 
-If the artifacts are too large for our NuGet feeds to handle, we can:
-
-1. Split the monster nupkg along some logical line, and produce multiple for a
-   single repo.
-2. Split some artifacts to a tarball that is published to an auxiliary feed.
-   * Extra infra development necessary: secure transport of this tarball,
-     implementing caching, etc.
-   * Not recommended.
+Some repositories produce artifacts that would be too large for Azure DevOps
+feeds to handle if put into a single intermediate package. To solve this, some
+repos need to produce "supplemental intermediate nupkgs".
+
+A "supplemental intermediate nupkg" uses the same naming convention as
+intermediate nupkgs, but inserts an extra name between the repo name and the
+RID.
+
+* `Microsoft.SourceBuild.Intermediate.runtime.libraries.linux.x64/6.0.0-foo`
+  * Contains the dotnet/runtime platform extensions libraries.
+* `Microsoft.SourceBuild.Intermediate.runtime.Microsoft.NETCore.App.Crossgen2.linux.x64/6.0.0-foo`
+  * Contains the crossgen2 pack nupkg.
+  * `Microsoft.NETCore.App.Crossgen2` is the name of the crossgen2 pack without
+    its RID, so use that as the name by convention.
+* `Microsoft.SourceBuild.Intermediate.runtime.Microsoft.NETCore.App.Crossgen2.archive.linux.x64/6.0.0-foo`
+  * This stores the `tar.gz` version of the crossgen2 pack.
+  * Crossgen2 is huge (~167 MB), so we need one supplemental nupkg for each
+    artifact--the nupkg and the archive.
+  * The name should apply to any RID, so I use `archive` rather than `tarball`
+    to allow Windows support in the future.
+
+Each nupkg should either be < 20 MB, or contain only one artifact. This is a
+heuristic to use to decide when to produce supplemental nupkgs and how to split
+them up.
+
+The intermediate nupkg we produce now (non-supplemental) contains:
+
```

Comment:
Supplemental nupkgs will just contain artifacts?

Reply:
Yep. Since the "main" package and all supplementals are built at the same time inside an official build, there's no data that's really unique to the supplemental package except the artifact itself. The "main" package is always restored and carries the reports (and list of supplemental packages).